src/protocol/messages/delete_records.rs

```diff
@@ -0,0 +1,294 @@
+    /// The topic name.
+    pub name: String_,
+
+    /// Each partition that we want to delete records from.
+    pub partitions: Vec<DeletePartitionRequest>,
+
+    /// The tagged fields.
+    ///
+    /// Added in version 2
+    pub tagged_fields: Option<TaggedFields>,
+}
+
+impl<W> WriteVersionedType<W> for DeleteTopicRequest
+where
+    W: Write,
+{
+    fn write_versioned(
+        &self,
+        writer: &mut W,
+        version: ApiVersion,
+    ) -> Result<(), WriteVersionedError> {
+        let v = version.0 .0;
+        assert!(v <= 2);
+
+        if v >= 2 {
+            CompactStringRef(&self.name.0).write(writer)?
+        } else {
+            self.name.write(writer)?;
+        }
+
+        if v >= 2 {
+            write_compact_versioned_array(writer, version, Some(self.partitions.as_slice()))?;
+        } else {
+            write_versioned_array(writer, version, Some(self.partitions.as_slice()))?;
+        }
+
+        if v >= 2 {
+            match self.tagged_fields.as_ref() {
+                Some(tagged_fields) => {
+                    tagged_fields.write(writer)?;
+                }
+                None => {
+                    TaggedFields::default().write(writer)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct DeleteRecordsRequest {
+    /// Each topic that we want to delete records from.
+    pub topics: Vec<DeleteTopicRequest>,
+
+    /// How long to wait for the deletion to complete, in milliseconds.
+    pub timeout_ms: Int32,
+
+    /// The tagged fields.
+    ///
+    /// Added in version 2
+    pub tagged_fields: Option<TaggedFields>,
+}
+
+impl<W> WriteVersionedType<W> for DeleteRecordsRequest
+where
+    W: Write,
+{
+    fn write_versioned(
+        &self,
+        writer: &mut W,
+        version: ApiVersion,
+    ) -> Result<(), WriteVersionedError> {
+        let v = version.0 .0;
+        assert!(v <= 2);
+
+        if v >= 2 {
+            write_compact_versioned_array(writer, version, Some(self.topics.as_slice()))?;
+        } else {
+            write_versioned_array(writer, version, Some(self.topics.as_slice()))?;
+        }
+        self.timeout_ms.write(writer)?;
+
+        if v >= 2 {
+            match self.tagged_fields.as_ref() {
+                Some(tagged_fields) => {
+                    tagged_fields.write(writer)?;
+                }
+                None => {
+                    TaggedFields::default().write(writer)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl RequestBody for DeleteRecordsRequest {
+    type ResponseBody = DeleteRecordsResponse;
+
+    const API_KEY: ApiKey = ApiKey::DeleteRecords;
+
+    /// All versions.
+    const API_VERSION_RANGE: ApiVersionRange =
+        ApiVersionRange::new(ApiVersion(Int16(0)), ApiVersion(Int16(2)));
+
+    const FIRST_TAGGED_FIELD_IN_REQUEST_VERSION: ApiVersion = ApiVersion(Int16(2));
+}
+
+#[derive(Debug)]
+pub struct DeletePartitionResponse {
```

Comment:
Same as the requests I think these should be named to make clear that they correspond to the DeleteRecords API